### PR TITLE
Fix compiler warnings in test_suite_pk.function

### PR DESCRIPTION
**Pull request by eleuzi01**
## Description

A fix for compiler warnings in test_suite_pk.function when SECP192R1 or/and SECP192K1 are disabled, see #9029 

## PR checklist

Please tick as appropriate and edit the reasons (e.g.: "backport: not needed because this is a new feature")

- [x] **changelog** done
- [x] **3.6 backport** done #9274
- [x] **2.28 backport** not required
- [x] **tests** not required

## Notes for the submitter

Please refer to the [contributing guidelines](https://github.com/Mbed-TLS/mbedtls/blob/development/CONTRIBUTING.md), especially the
checklist for PR contributors.

Help make review efficient:
* Multiple simple commits
  - please structure your PR into a series of small commits, each of which does one thing
* Avoid force-push
  - please do not force-push to update your PR - just add new commit(s)
* See our [Guidelines for Contributors](https://mbed-tls.readthedocs.io/en/latest/reviews/review-for-contributors/) for more details about the review process.

### CODE REVIEW

**Review by sezrab — CHANGES_REQUESTED** [review]
Looks good. Should it have a change-log since these are user-facing changes?

**Review by minosgalanakis — DISMISSED** [review]
LGTM

**Comment by minosgalanakis** [comment]
> Looks good. Should it have a change-log since these are user-facing changes?

We do not normally do changelog entries for test code.

We **should have** 3.6 backports though.

**Review by sezrab — DISMISSED** [review]

**Comment by gilles-peskine-arm** [comment]
We don't write changelog entries for test improvements, but this is a build failure, and it's user-reported, so I think we should have a changelog entry.

**Comment by eleuzi01** [comment]
> We don't write changelog entries for test improvements, but this is a build failure, and it's user-reported, so I think we should have a changelog entry.

Done, let me know if it's good.

**Review by gilles-peskine-arm — DISMISSED** [review]
LGTM

File: ChangeLog.d/fix-test-suite-pk-warnings.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Bugfix
+   * Fix redefinition warnings when SECP192R1 and/or SECP192K1 are disabled.
+     Fixes #9029.
```
Comment (gilles-peskine-arm) [inline ChangeLog.d/fix-test-suite-pk-warnings.txt:3]: The last line is missing a newline at the end. That confuses some tools including `assemble_changelog.py`.

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

**Review by minosgalanakis — APPROVED** [review]
LGTM

**Comment by paul-elliott-arm** [comment]
Ignoring Open CI fail, as internal passed.

**Pull request merged**